File: JK/222. Count Complete Tree Nodes.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

def countNodes(root):
    
    def dfs(node):
        if not node:
            return 0
        return 1 + dfs(node.left) + dfs(node.right)

    return dfs(root)

File: JK/222. Count Complete Tree Nodes.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
#include <stdio.h>
#include <stdlib.h>

// Function prototype for counting nodes
int countNodes(struct TreeNode* root);

// Recursive function to count nodes
int dfs(struct TreeNode* node) {
    if (!node) {
        return 0;
    }
    return 1 + dfs(node->left) + dfs(node->right);
}

int countNodes(struct TreeNode* root) {
    return dfs(root);
}